src/istio/mixerclient/client_impl.cc

```diff
@@ -117,11 +125,16 @@ CancelFunc MixerClientImpl::Check(
        on_done](const Status &status) {
         raw_check_result->SetResponse(status, *request_copy, *response);
         raw_quota_result->SetResponse(status, *request_copy, *response);
+        CheckResponseInfo check_response_info;
+        check_response_info.is_check_cache_hit = false;
+        check_response_info.is_quota_cache_hit = false;
         if (on_done) {
           if (!raw_check_result->status().ok()) {
-            on_done(raw_check_result->status());
+            check_response_info.response_status = raw_check_result->status();
+            on_done(check_response_info);
           } else {
-            on_done(raw_quota_result->status());
+            check_response_info.response_status = raw_quota_result->status();
+            on_done(check_response_info);
```

Comment:
combine above,  to only call on_done() once,   just setting status is different

Reply:
Done.